### DISCUSSION TITLE
Collision between the ImageMagick version (6.5.4.10) of an installed application and the Magick.net 7.22.2.0 version of my project.

### DISCUSSION THREAD

**Discussion by pherrera87** [dlemstra/Magick.NET#798] [Help]
I am using Zbarimg.exe I have seen that it includes ImageMagick version 6.5.4-10, but my application uses ImageMagick.net version 7.22.2.0 by default and it seems that these are colliding.
For example, I use the zbarimg to read a barcode from the console and then I convert to PDF format from my app, the first time it works but if I run this process a second time my Magick.net will have no effect.

Any idea what could be happening?

**Comment by dlemstra** [comment]
Maybe you can get some information from debug logging (see documentation on how to enable that). It might be possible that the code of `Zbarimg.exe` is setting some environment variables that are used by ImageMagick. And that could cause the code of ImageMagick to fail. If you are using the AnyCPU version I would advise you to switch to either the x64 (preferred) or the x86 version of the library.

**Comment by pherrera87** [comment]
Hi,

I have fixed it by clearing the environment variables when I run the command within the process.

use  proc.StartInfo.EnvironmentVariables.Clear();

```
Process proc = new Process();
            proc.StartInfo.FileName = barcodeEnginePath;
            proc.StartInfo.Arguments = ars;
            proc.StartInfo.CreateNoWindow = true;
            proc.StartInfo.WindowStyle = ProcessWindowStyle.Hidden;
            proc.StartInfo.UseShellExecute = false;
            proc.StartInfo.RedirectStandardOutput = true;
            proc.StartInfo.EnvironmentVariables.Clear();
            proc.Start();
```
Regards.
